api/pom.xml

```diff
@@ -329,7 +329,7 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>
                 <configuration>
-                    <argLine>@{argLine} --illegal-access=permit</argLine>
```

Comment:
can you elaborate on this one please?

Reply:
Yes, in deed I should have written something about it in the PR description. 
In the maven build there was this warning : 
`OpenJDK 64-Bit Server VM warning: Ignoring option --illegal-access=permit; support was removed in 17.0`
As the project is now using Java 21, I removed it. 
do you remember why this option was added?

Reply:
not quite, let's see if it works. Doesn't matter otherwise, since we're migrating to gradle which doesn't have surefire